parachain/pallets/basic-channel/src/inbound/envelope.rs

```diff
@@ -20,6 +21,8 @@ static EVENT_ABI: &Event = &Event {
 pub struct Envelope {
 	/// The address of the outbound channel on Ethereum that forwarded this message.
 	pub channel: H160,
+	/// The user on Ethereum that authorized the source to send the message.
+	pub origin: H160,
```

Comment:
In this context, lets rather name this `user`.

Also, we can switch the order of `source` and `user`, so that user is grouped with nonce?